core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -205,6 +211,8 @@ public <T> T get(int pos, Class<T> javaClass) {
   public <T> void set(int pos, T value) {
     switch (pos) {
       case 0:
+        Preconditions.checkArgument(
```

Comment:
It feels a bit awkward to rely on `Record` for a nested field while the main object is simply `StructLike`.

Reply:
I can keep the member as `Record` instead of `StructLike` to avoid this. But since we have plan to use internal writers in future (which uses `StructLike`), we lose compatibility if we keep the member as `Record` instead of `StructLike`.

I don't think it is too awkward as `Record` implements `StructLike`.